hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -206,18 +213,23 @@ public Target updateTarget(final TargetUpdate u) {
     @Override
     @Modifying
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
-    public void deleteTargets(final Long... targetIDs) {
-        deleteTargets(Lists.newArrayList(targetIDs));
+    public void deleteTargets(final Collection<Long> targetIDs) {
+        targetRepository.deleteByIdIn(targetIDs);
+
+        targetIDs.forEach(targetId -> eventPublisher.publishEvent(new TargetDeletedEvent(tenantAware.getCurrentTenant(),
```

Comment:
Do you know why it is necessary to fire the deletion event?
Normally the `JpaTarget#fireDeleteEvent` should fire the event.